deltachat-ffi/src/lib.rs

```diff
@@ -1818,3 +1863,26 @@ fn as_opt_str<'a>(s: *const libc::c_char) -> Option<&'a str> {
 
     Some(dc_tools::as_str(s))
 }
+
+pub trait ResultExt<T: Default> {
+    fn unwrap_or_log_default(self, context: &context::Context, message: &str) -> T;
+    fn log_err(&self, context: &context::Context, message: &str);
```

Comment:
Would it not be better if this passed along the Result?  I think it's more flexible in use.  Also unwap_or_log_default could be implemented using it in that case.